### PR TITLE
implementation of Sing in service, controller, tests and documentation

### PR DESCRIPTION
**Pull request by DEG-W**
## Summary

Implementation of Sing in service, controller, tests and documentation

## Known Issues

missing log an test error in case of database failure

## Trello Card

https://trello.com/c/j2f2Nvmm

### CODE REVIEW

File: app/Utiles/users.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const bcrypt = require('bcryptjs');
+const jwt = require('jwt-simple');
+const { jwtSecret } = require('../../config').common.jwt;
+
+exports.findUserByEmail = (model, email) => model.findOne({ where: { email } });
```
Comment (marioportela1993) [inline app/Utiles/users.js:5]: This function is used in the users service. Maybe you can put this function in the `/services/users.js` file. On the other hand, you can rename the file as token.js o something similar, due to all functions here are used to manage the token.

File: app/Utiles/users.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const bcrypt = require('bcryptjs');
```
Comment (marioportela1993) [inline app/Utiles/users.js:1]: Rename the folder as `utils`, don't use `Utiles`,  i.e., change `app/Utiles/` to  `app/utils/`

File: app/controllers/users.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-const { createUserService } = require('../services/users');
+const { createUserService, singInUserService } = require('../services/users');
 const { users } = require('../models');
```
Comment (marioportela1993) [inline app/controllers/users.js:2]: Call models into services

File: app/services/users.js [hunk 4]
```diff
@@ -1,20 +1,30 @@
...
+  findUserByEmail(users, data.email)
+    .then(existUser => {
+      if (!existUser) throw unauthenticatedUserError(`user with email ${data.email} is not registred`);
+      return [checkPass(data.password, existUser.password), existUser.id];
+    })
+    .then(([checkedPass, userId]) => {
+      if (checkedPass) return generateUserJWT({ id: userId });
+      throw unauthenticatedUserError('unauthenticated user');
+    })
+    .catch(error => {
```
Comment (marioportela1993) [inline app/services/users.js:27]: Here you are logged whatever error occurred when you do database operations. 🥇

File: app/services/users.js [hunk 5]
```diff
@@ -1,20 +1,30 @@
...
+      logger.error(error);
+      throw error;
```
Comment (marioportela1993) [inline app/services/users.js:29]: Use a custom error here, you can use the `databaseError` that already exist in the `errors.js` file. Example: `throw databaseError('Database error when trying  to singing in  an user')`

File: config/index.js [hunk 6]
```diff
@@ -47,6 +47,9 @@ const config = {
     },
     service: {
       numberAPIUrl: process.env.URL_NUMBERS_API
+    },
+    jwt: {
```
Comment (marioportela1993) [inline config/index.js:51]: You can use the `secret` in the `session` key above

File: test/controllers/usere.spec.js [hunk 7]
```diff
@@ -29,34 +30,75 @@ describe('controllers', () => {
...
           }));
     });
+    describe('POST /users/sessions', () => {
+      it('Sing in with a registered user', () =>
+        userFactory
+          .create()
+          .then(() =>
+            request(app)
+              .post('/users/sessions')
+              .send({ email: user.email, password: user.password })
```
Comment (marioportela1993) [inline test/controllers/usere.spec.js:69]: Change `.send({ email: user.email, password: user.password })` to `.send(user)` is the same. What do you think?

File: test/controllers/usere.spec.js [hunk 8]
```diff
@@ -29,34 +30,75 @@ describe('controllers', () => {
...
+          )
+          .then(res => {
+            expect(res.statusCode).toBe(VALID_RESPONSE_CODE);
+          }));
+      it('Sing in with a unregistered user', () =>
+        request(app)
+          .post('/users/sessions')
+          .send({ email: user.email, password: user.password })
```
Comment (marioportela1993) [inline test/controllers/usere.spec.js:77]: Same comment

File: test/controllers/usere.spec.js [hunk 9]
```diff
@@ -29,34 +30,75 @@ describe('controllers', () => {
...
+          .then(res => {
+            expect(res.statusCode).toBe(UNAUTHENTICATED_USER_ERROR);
+          }));
+      it('Sing in user with invalid domain email', () =>
+        request(app)
+          .post('/users')
+          .send({ email: 'pedro.perez@gmail.com', password: '12345678' })
```
Comment (marioportela1993) [inline test/controllers/usere.spec.js:84]: Change `.send({ email: 'pedro.perez@gmail.com', password: '12345678' })` to `.send({ ...user, password: '12345678' })`, is the same and more beautiful  :)

File: test/controllers/usere.spec.js [hunk 10]
```diff
@@ -29,34 +30,75 @@ describe('controllers', () => {
...
+          .then(res => {
+            expect(res.statusCode).toBe(INVALID_SCHEMA_CODE);
+          }));
+      it('Sing in user with short password', () =>
+        request(app)
+          .post('/users')
+          .send({ email: 'pedro.perez@.wolos.com', password: '123456' })
```
Comment (marioportela1993) [inline test/controllers/usere.spec.js:91]: Same comment

File: test/controllers/usere.spec.js [hunk 11]
```diff
@@ -29,34 +30,75 @@ describe('controllers', () => {
...
+          .then(res => {
+            expect(res.statusCode).toBe(INVALID_SCHEMA_CODE);
+          }));
+      it('Sing in user with no alphanumeric password', () =>
```
Comment (marioportela1993) [inline test/controllers/usere.spec.js:95]: Same comment

**Review by marioportela1993 — CHANGES_REQUESTED** [review]

File: app/services/users.js (hunk 4 above)
Comment (DEG-W) [reply app/services/users.js:27]: I don't know how to identify the database error to separate it from the others

**Review by marioportela1993 — APPROVED** [review]

**Pull request merged**